internal/node/node.bzl

```diff
@@ -40,27 +40,19 @@ def _trim_package_node_modules(package_name):
         segments.append(n)
     return "/".join(segments)
 
-def _compute_node_modules_root(ctx):
-    """Computes the node_modules root from the deps attributes.
-
-    Args:
-      ctx: the skylark execution context
-
-    Returns:
-      The node_modules root as a string
-    """
-    node_modules_root = None
+def _compute_node_modules_roots(ctx):
+    """Computes the node_modules root (if any) from data attribute."""
+    node_modules_roots = {}
     for d in ctx.attr.data:
         if ExternalNpmPackageInfo in d:
-            possible_root = "/".join([d[ExternalNpmPackageInfo].workspace, "node_modules"])
-            if not node_modules_root:
-                node_modules_root = possible_root
-            elif node_modules_root != possible_root:
-                fail("All npm dependencies need to come from a single workspace. Found '%s' and '%s'." % (node_modules_root, possible_root))
-    if not node_modules_root:
-        # there are no fine grained deps but we still need a node_modules_root even if it is a non-existant one
-        node_modules_root = "build_bazel_rules_nodejs/node_modules"
-    return node_modules_root
+            path = d[ExternalNpmPackageInfo].path
+            workspace = d[ExternalNpmPackageInfo].workspace
+            if path in node_modules_roots:
+                other_workspace = node_modules_roots[path]
+                if other_workspace != workspace:
+                    fail("All npm dependencies at the path '%s' must to come from a single workspace. Found '%s' and '%s'." % (path, other_workspace, workspace))
```

Comment:
"must to come" again - I guess this is a copy pasta we ought to clean up

Reply:
🌮